ui/test/services/project_service_test.dart

```diff
@@ -36,20 +36,32 @@ void main() {
     });
 
     group('Projects query', () {
-      test('queries list of projects', () async {
-        when(client.getProjects())
-            .thenAnswer((_) => Future.value([Project(id: projectId)]));
+      test('query to list all projects when no filter applied', () async {
+        when(client.findProjectsBySearchFragment(''))
+            .thenAnswer((_) => Future.value([
+                  Project(id: projectId, title: 'ProjectA'),
+                  Project(id: projectId, title: 'ProjectB')
+                ]));
 
-        final projects = await service.allProjects();
+        final projects = await service.findProjects('');
+
+        expect(projects, hasLength(2));
+      });
+
+      test('query to list only one filtered project - ProjectB', () async {
```

Comment:
I try to name tests after the expected behaviour of the object under test, which leads to names like e.g. "queries by fragment".

I don't think "ProjectB" is relevant to the behaviour of the service, so my suggestion is to therefore not mention it in the title.